src/app/scheduler/components/CourseCard.tsx

```diff
@@ -0,0 +1,92 @@
+import { AddIcon, CloseIcon, InfoOutlineIcon } from '@chakra-ui/icons';
+import { Box, Text, Flex, VStack, Checkbox, IconButton, Link } from '@chakra-ui/react';
+import { useCallback } from 'react';
+
+import { Term } from '../../../shared/fetchers';
+import { useSavedCourses } from '../../../shared/hooks/useSavedCourses';
+
+export type CourseCardProps = {
+  term: Term;
+  subject: string;
+  title: string;
+  code: string;
+  colour: string;
+  pid: string;
+  selected: boolean;
+  handleChange: Function;
+};
+
+export function CourseCard({
+  term,
+  subject,
+  title,
+  code,
+  colour,
+  pid,
+  handleChange,
+  selected,
+}: CourseCardProps): JSX.Element {
+  const { addCourse, deleteCourse, contains } = useSavedCourses();
+  const courseIsSaved = contains(pid, term);
+
+  const onCourseClick = useCallback(() => {
+    if (term && pid !== '' && code !== '') {
+      if (!courseIsSaved) {
+        addCourse({ subject, code, pid, term });
+      } else {
+        deleteCourse({ subject, code, pid, term });
+      }
+    }
+  }, [term, pid, code, courseIsSaved, addCourse, subject, deleteCourse]);
```

Comment:
I think this can be removed and the component using `CourseCard` can be responsible for this stuff.